# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1329507627 28800
#      Fri Feb 17 11:40:27 2012 -0800
# Node ID 81183cd6eaae7589dae484dcab73eb3fd84474df
# Parent  dbc99d17f0add4d32f9648d4b737bc8f07b1635b
Fix OpenGL printing of dashed lines (Bug #33765).

* gl2ps-renderer.h (set_linestyle): Fix OpenGL printing of dashed lines (Bug #33765)

diff --git a/src/gl2ps-renderer.h b/src/gl2ps-renderer.h
--- a/src/gl2ps-renderer.h
+++ b/src/gl2ps-renderer.h
@@ -47,23 +47,24 @@ protected:
 
 
   void set_font (const base_properties& props);
 
   void draw_text (const text::properties& props);
   void draw_pixels (GLsizei w, GLsizei h, GLenum format,
                     GLenum type, const GLvoid *data);
 
-  void set_linestyle (const std::string& s, bool use_stipple)
+  void set_linestyle (const std::string& s, bool use_stipple = false)
   {
     opengl_renderer::set_linestyle (s, use_stipple);
-    if (use_stipple)
+
+    if (s == "-" && ! use_stipple)
+      gl2psDisable (GL2PS_LINE_STIPPLE);
+    else
       gl2psEnable (GL2PS_LINE_STIPPLE);
-    else
-      gl2psDisable (GL2PS_LINE_STIPPLE);
   }
 
   void set_polygon_offset (bool on, double offset = 0.0)
   {
     opengl_renderer::set_polygon_offset (on, offset);
     if (on)
       gl2psEnable (GL2PS_POLYGON_OFFSET_FILL);
     else
